spark/src/main/scala/geotrellis/spark/pyramid/Pyramid.scala

```diff
@@ -98,7 +112,12 @@ object Pyramid extends LazyLogging {
         .map { case (key, tile) =>
           val extent: Extent = key.getComponent[SpatialKey].extent(sourceLayout)
           val newSpatialKey = nextLayout.mapTransform(extent.center)
-          (key.setComponent(newSpatialKey), (key, tile))
+          // Resample the tile on the map side of the pyramid step.
+          // This helps with shuffle size.
+          val resampled = tile.prototype(nextLayout.tileLayout.tileCols, nextLayout.tileLayout.tileRows)
```

Comment:
I see this line so often, can we implement some function that would incapsulate this merge into prototype logic?